fix(ci): contracts build で hardhat-abi-exporter が ABI を生成しない

## 概要

CI の `Build (22.x)` で `@niji/contracts#build` が ABI 不在で fail する。
`pnpm build:sol`(hardhat compile + hardhat-abi-exporter) が正常終了したはずなのに、 直後の `pnpm build:ts`(tsup) が `src/index.ts` から `../abi/contracts/**.json` を import できず TS2307 で死ぬ。
CAR-301 (turbo subgraph→contracts 順序強制) を merge した PR #2985 後も continue failure 状態のため、 contracts build の ABI export 経路を修正する。

## 課題

- CI で `hardhat compile` が走った後でも `abi/contracts/` が空のままで tsup が落ちる
- local では `abi/contracts/` に 17 entry 存在し、 git untracked (`.gitignore` 対象) のため CI clone 時には不在
- `subgraph.yaml` / `contracts/src/index.ts` の両方が `../abi/...` を参照しているため、 abi 生成は両 package の build success の前提

## 詳細

エラー抜粋 (run 28420263511)。

```
✘ [ERROR] Could not resolve "../abi/contracts/NijiToken.sol/NijiToken.json"
src/index.ts:1:40: ERROR: Could not resolve "../abi/contracts/NijiToken.sol/NijiToken.json"
##[error]src/index.ts(1,41): error TS2307: Cannot find module '../abi/contracts/NijiToken.sol/NijiToken.json'
```

設定の現状。

| 項目 | 値 | 場所 |
|---|---|---|
| build script | `pnpm build:sol && pnpm build:ts` | `packages/niji-contracts/package.json:21` |
| build:sol | `hardhat compile` | `package.json:22` |
| build:ts | `tsup` | `package.json:23` |
| abiExporter | `{ path: './abi', clear: true }` | `hardhat.config.ts:87-90` |
| tsup loader | `{ '.json': 'json' }` | `tsup.config.ts:23-25` |
| import 元 | `../abi/contracts/*.sol/*.json` (7 ABI) | `src/index.ts:1-7` |

```mermaid
graph LR
    A[pnpm build:sol] --> B[hardhat compile]
    B --> C[hardhat-abi-exporter]
    C -.->|期待| D[abi/contracts/*.json 生成]
    A --> E[pnpm build:ts]
    E --> F[tsup]
    F -->|src/index.ts import 解決| D
    D -->|CI で生成されず| G[TS2307 fail]
```

仮説。

1. `hardhat-abi-exporter` が compile flow に hook せず、 明示的に export task が必要
2. `clear: true` で先にディレクトリを消すが、 generation 経路が壊れている
3. solc compiler download (CI で `Downloading compiler 0.8.23`) が遅延し、 compile 自体が completed していない可能性

## 解決方法

3 手段を順に試す。

1. `hardhat-abi-exporter` の export task を明示的に呼ぶ (`hardhat export-abi` を build:sol に追加)
2. `pretest` パターンで `hardhat compile --force` を強制実行
3. abi 生成を build:sol 内で `pnpm hardhat compile && pnpm hardhat export-abi` の 2 step 化

## 仕組み

```mermaid
graph LR
    A[pnpm build:sol] --> B[hardhat compile]
    B --> C[hardhat export-abi 明示呼出]
    C --> D[abi/contracts/*.json 生成保証]
    D --> E[pnpm build:ts]
    E --> F[tsup loader json で取込]
    F --> G[dist/index.js success]
```

## 進め方

### Phase 1 — 調査・方針決定
- [ ] T001 local で `rm -rf abi && pnpm build:sol` を実行し ABI が再生成されるか検証 `packages/niji-contracts/`
- [ ] T002 `pnpm hardhat export-abi` cmd が単独で動くか確認 (hardhat-abi-exporter docs)
- [ ] T003 fail run 28420263511 詳細 log で `hardhat compile` の exit code を確認

### Phase 2 — 実装
- [ ] T004 修正方針を 3 案 (export-abi 明示 / compile --force / clear: false) から決定
- [ ] T005 採用案を `packages/niji-contracts/package.json` の build:sol script に反映
- [ ] T006 必要なら `.gitignore` から `abi/` 除外を解除し、 ABI を commit 対象化 (alternative)

### Phase 3 — 検証・提出
- [ ] T007 local で `pnpm -F @niji/contracts build` が green
- [ ] T008 PR push 後 CI `Build (22.x)` で contracts + subgraph + sdk + webapp 全 green を確認
- [ ] T009 PR 作成

## タスク詳細

### T001 — local 再現手順

```bash
cd packages/niji-contracts
rm -rf abi typechain artifacts cache
pnpm build:sol
ls abi/contracts/  # 17 entry 出るはず
pnpm build:ts      # success すれば local では正常
```

CI で同じ手順が失敗するなら、 ① solc download が遅い ② abi-exporter が hook してない の 2 択。

## 完了条件

- [ ] master の `Build (22.x)` CI が green (contracts + subgraph 共に build success)
- [ ] `packages/niji-contracts/abi/contracts/` に 17 entry が CI build 後に生成される
- [ ] `@niji/contracts#build` が turbo cache miss でも安定 success
- [ ] subgraph build も連動して success

## 影響範囲

- `packages/niji-contracts/package.json` (build script 修正 1-2 行)
- `packages/niji-contracts/hardhat.config.ts` (abiExporter 設定変更可能性)
- 可能性として `.gitignore` (abi/ commit 化方針なら)
- CI build 時間 (compile + abi export で +20-30 秒見込み)

## 関連

- PR #2985 (CAR-301 / turbo subgraph→contracts 配線、 本 Issue の前段)
- PR #2984 (mount-unmount revert、 master CI failure 発見の契機)
- CAR-301 (turbo 配線漏れ)
- 失敗 run = https://github.com/Niji-DAO/niji-monorepo/actions/runs/28420263511

--- a/packages/niji-contracts/hardhat.config.ts
+++ b/packages/niji-contracts/hardhat.config.ts
@@ -87,6 +87,7 @@ const config: HardhatUserConfig = {
   abiExporter: {
     path: './abi',
     clear: true,
+    runOnCompile: true,
   },
   typechain: {
     outDir: './typechain',

